src/prepare/mod.rs

```diff
@@ -154,7 +154,7 @@ pub fn deny_floating_point(module: &Module) -> Result<(), Error> {
                         if func
                             .params()
                             .iter()
-                            .chain(func.return_type().as_ref())
+                            .chain(func.results().first())
```

Comment:
Ah, missed this one. Here we check that none of the function parameters have a FP type, and here it checks only the first result. While it's ok-ish since wasmi will fail on validation of multivalue result types, it would be still good to fix.

Reply:
Do you mean we should then chain the whole `results()` iterator?

Reply:
Related to the same change for return types, shall the `Signature` be changed to reflect that? Instead of only picking the first type in `from_element` in `src/types.rs`.

Reply:
Yes, we should go over all the result types, not just the first one.

The signature should be left as is since wasmi doesn't support the multivalue proposal as of now. As I mentioned above, multivalue code will be rejected anyway

Reply:
Should now be fixed.